src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -119,7 +119,7 @@
     private SeverityLevel severity = SeverityLevel.ERROR;
 
     /** Name of a charset. */
-    private String charset = System.getProperty("file.encoding", StandardCharsets.UTF_8.name());
+    private String charset = StandardCharsets.UTF_8.name();
```

Comment:
we need to update xdoc.
https://checkstyle.org/config.html#Checker_Properties

This update is breaking compatibility change .... we need to be clear with users on this.

Reply:
So, should I change the example to another charset?  I will also correct the table here: https://github.com/checkstyle/checkstyle/blob/834b505b9938e93f6950dd20a13e39e295e5ac86/src/xdocs/config.xml#L340

to:
`              <td><code>UTF-8</code></td>
`

Reply:
> So, should I change the example to another charset?

yes, you can use bulgarian charset as it is in issue. 

>  will also correct the table here

no, it is version for when property appear.

Reply:
Fixed.